### PR TITLE
Graylog 6.0: update configuration script

### PR DESCRIPTION
**Pull request by YuryHrytsuk**
## What do these changes do?
* Fix long-standing issue with waiting for graylog to be online (we did not wait enough)
* Switch from index rotation (deprecated in Graylog 6.0) to data tiering
* Add Graylog major version check to the script

## Related PR/s
* https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/merge_requests/1002

## Checklist
- [X] I tested and it works

### CODE REVIEW

File: services/graylog/scripts/configure.py [hunk 1]
```diff
@@ -380,21 +390,13 @@ def install_content_pack_revision(content_pack):
...
-        print("Graylog script will now stop.")
-        sys.exit(1)
+    wait_graylog_is_online()
+    validate_graylog_version_is_supported()
 
     session = requests.Session()
     session.verify = False
     session.auth = (
-        "admin",
+        env.str("SERVICES_USER"),
```
Comment (mrnicegyu11) [inline services/graylog/scripts/configure.py:399]: nice!

File: services/graylog/scripts/configure.py [hunk 2]
```diff
@@ -121,54 +139,46 @@ def configure_email_notifications(_session, _headers):
...
+    )
+    _r = _session.get(_url, headers=_headers, verify=False)
+    index_of_interest = [
+        index
+        for index in _r.json()["index_sets"]
+        if index["title"] == "Default index set"
+    ][0]
+
+    # https://graylog.org/post/understanding-data-tiering-in-60-seconds/
+    # https://community.graylog.org/t/graylog-6-0-data-tiering-for-rotation-and-retention/33302
```
Comment (mrnicegyu11) [inline services/graylog/scripts/configure.py:155]: nice!

File: services/graylog/scripts/configure.py [hunk 3]
```diff
@@ -25,31 +31,44 @@
...
+SUPPORTED_GRAYLOG_MAJOR_VERSION = 6
+
+GRAYLOG_BASE_DOMAIN = "https://monitoring." + env.str("MACHINE_FQDN") + "/graylog"
+GRAYLOG_WAIT_ONLINE_TIMEOUT_SEC = env.int("GRAYLOG_WAIT_ONLINE_TIMEOUT_SEC")
+REQUESTS_AUTH = (env.str("SERVICES_USER"), env.str("SERVICES_PASSWORD"))
 
-def log_attempt_number(retry_state):
-    """return the result of the last call attempt"""
-    print(f"Retrying: {retry_state.attempt_number}...")
+GRAYLOG_LOG_MAX_DAYS_IN_STORAGE = env.int("GRAYLOG_LOG_MAX_DAYS_IN_STORAGE")
```
Comment (mrnicegyu11) [inline services/graylog/scripts/configure.py:40]: nice!

**Review by mrnicegyu11 — APPROVED** [review]
thanks for handling this!

**Comment by mrnicegyu11** [comment]
I guess there is a PR introducing those envs in osparc-config?

**Comment by YuryHrytsuk** [comment]
> I guess there is a PR introducing those envs in osparc-config?

Yeah, check the `Related PRs` in this issue's body

**Pull request merged**